hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/SevenToEightUpgradeHandler.java

```diff
@@ -45,29 +40,6 @@ public Map<ConfigProperty, String> upgrade(HoodieWriteConfig config, HoodieEngin
     final HoodieTableConfig tableConfig = upgradeDowngradeHelper.getTable(config, context).getMetaClient().getTableConfig();
 
     Map<ConfigProperty, String> tablePropsToAdd = new HashMap<>();
-    if (tableConfig.getTableType().equals(HoodieTableType.MERGE_ON_READ)) {
```

Comment:
We still need to keep the logic of adding the required table configs in table version 8, correct?  We should avoid the Hudi 1.0 writer to change table configs as much as possible, i.e., invariant table configs should either be created for a new table or changed during upgrade/downgrades, except for MDT related table configs, to avoid possible corruption.

Reply:
Let's create a follow-up Jira ticket to fix the table upgrade and downgrade around the record merge configs.

Reply:
https://issues.apache.org/jira/browse/HUDI-8409